File: 03_Arreglos/Matrices/CSharp/ejemplo2.cs
using System;
class Programa {
    static void Main() {
        Console.Write("n (para matriz n x n): ");
        int n = int.Parse(Console.ReadLine() ?? "0");

        int[,] M = new int[n, n];
        Console.WriteLine("\nIngrese elementos de M:");
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++) {
                Console.Write($"M[{i},{j}] = ");
                M[i, j] = int.Parse(Console.ReadLine() ?? "0");
            }

        int[,] T = new int[n, n];
        int sumaDiag = 0;

        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++) {
                T[j, i] = M[i, j];
                if (i == j) sumaDiag += M[i, j];
            }

        Console.WriteLine("\nTranspuesta T:");
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++)
                Console.Write(T[i, j].ToString().PadLeft(6));
            Console.WriteLine();
        }

        Console.WriteLine($"\nSuma de la diagonal principal = {sumaDiag}");
    }
}

File: 03_Arreglos/Matrices/Python/ejemplo2.py
def transpuesta(M):
    return [list(fila) for fila in zip(*M)]

def suma_diagonal(M):
    return sum(M[i][i] for i in range(len(M)))

def main():
    n = int(input("n (para matriz n x n): "))
    M = []
    print("\nIngrese elementos de M:")
    for i in range(n):
        fila = []
        for j in range(n):
            fila.append(int(input(f"M[{i},{j}] = ")))
        M.append(fila)

    T = transpuesta(M)
    diag = suma_diagonal(M)

    print("\nTranspuesta T:")
    for fila in T:
        print("".join(str(x).rjust(6) for x in fila))

    print(f"\nSuma de la diagonal principal = {diag}")

if __name__ == "__main__":
    main()
